### ISSUE TITLE
Question about Atom vs. Linux

### ISSUE THREAD

**Issue by juanpc2018** [atom/atom#25589]
i noticed a common problem across all linux builds....
also on OSX

for example, 
a code is designed using Ubuntu 17,
or OSX SnowLeopard 

the same code becomes deprecated in 20.04.4 LTS,
and super deprecated in 22.04 LTS
or OSX Mavericks, or HighSierra, or Catalina.

gives errors about deprecated functions...
you should replace "this" with "this"...

when compiling on a different linux or OSX. 
some code becomes broken and impossible to build.

problem #2.
most developers have a pre-installed machine 
with lots of libraries, most people does Not have exactly.

when you try to compile again on a clean machine, 
some codes are impossible to build,
or builds with lots of warnings.

most developers never test compiling on a clean machine.
OS changes make impossible to install some libraries on a clean install.

my question is:
Atom "now owned by Microsoft"
VScode "owned by Microsoft" but open source. 

¿has a compatibility translator feature?

for example:
Input vs. output

Input version code:
[Ubuntu 17.04] 

Ouput:
[Ubuntu 17.10]
or
20.04 LTS
or 
22.04 LTS
or 18.04
or 18.10
or 20.04
or 20.10
or 10.10
or 10.04

Atom or VSstudio translate all deprecated functions automatically.
Ubuntu 17 to 20,
or 22 to 20
or 17 to 16
or 17 to 18
18 to 20
20 to 21
21 to 22
etc...

to increase source compatibility..

i find too crazy to manually edit 10000 deprecated functions just to compile on a New OS, 
even with copy & paste is too much effort.

Github is becoming a graveyard of dead code.

this is too crazy.

**Comment by rilysh** [comment]
Yeah, but the issue isn't with the source code as JavaScript is an interpreted language and Atom uses Node, so wherever Node can run (v8 engine) JavaScript also can run.

All of these issues actually comes from Atom's dependencies, where most of them are deprecated or the OS doesn't provide any library support anymore. If you alsp noticed, Atom has quite a lot of dependencies, some of them it never uses (nested dependencies). It's just not ended but there's much more thing that can be done right.

For now, I'd suggest to use VSCode, it's actively maintained and comfortable with Atom (in most ways, and features etc). There's nothing much we can do about it and Atom wasn't even got any major update in years.
